infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/LRRollingUpgradeHandler.java

```diff
@@ -129,6 +129,10 @@ public boolean isLRUpgradeInProgress(TxnContext txnContext) {
      * @param txnContext All of the above must execute in the same transaction passed in.
      */
     public void migrateData(TxnContext txnContext) {
+        if(LogReplicationMetadataManager.isMigrationComplete(txnContext)) {
+            log.info("Data migration for LR V2 has already been completed.");
+            return;
+        }
```

Comment:
Prior to reaching migrateData could we use this check to set the `isClusterAllAtV2` value since that field not being persisted on restarts is the cause of the issue?

Reply:
Yeah, that is certainly a way to do, think its safe as well since LR doesn't start up until all the nodes are upgraded.
But having it here, means there are some real-time checks in the path to ensure that all the nodes are indeed on LR V2...So I am inclined to keep it here, I think its more safe incase the behaviour above changes in the future...

But I am open to change it if others feel safe to move this check up..